app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/logging_mixin.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# Encapsulate logging methods into a single mixin.
+module ManageIQ::Providers::IbmCloud::VPC::CloudManager::LoggingMixin
```

Comment:
The purpose of the logs and various methods is as follows:

- provider specific log (e.g. `$ibm_cloud_log`) - This purpose of this log is to have a separate logger for the low-level operations that are specific to the provider.  Typically this is just the request/response, and in many providers this logger is given directly to the SDK library for them to log to.  This allows a user to tune the verbosity of the "native" logger independently of the application logger.
- `$log` - This is the primary logger for the application.  Since this PR is about provisioning I would expect all provisioning things to go through this logger (except the low-level native calls).
- `_log` - This is a proxy method for `$log` provided by the [Vmdb::LogProxy](https://github.com/ManageIQ/manageiq/blob/4e7b8e882d6cc6ab224dfea770a3129438d5a82d/lib/vmdb/logging.rb#L4), and its only purpose is to prepend code location information into the `$log` entry in the format `MIQ(Class.method_name) - `.  Having the consistent format allows for readability and for diagnostics when we post-process the logs.  Ideally this would live in $log directly and we wouldn't need this logger, but Ruby doesn't have the capability to know the calling class and method names via the backtrace, so we needed this unfortunate proxy.

Additionally, on appliances, all logs are broadcast to journald and on containers/podified all logs are broadcast to STDOUT in JSON <sup>[[ref]](https://github.com/ManageIQ/manageiq/blob/4e7b8e882d6cc6ab224dfea770a3129438d5a82d/lib/vmdb/loggers.rb#L34-L38)</sup>.

So, given that, part of this class feels unnecessary because it's effectively wrapping a wrapper which already prepends code location information.  The other half seems to deal with the development complexity of having to watch logs in multiple files, but that feels like it can be handled in a simpler way.

I think this class should be removed in favor of using `$ibm_cloud_log` directly for native SDK calls and either `$log` or `_log` for the rest of the logging calls.

For the multiple logs problem I'd recommend doing nothing code-wise and locally just doing `tail -f log/evm.log log/ibm_cloud.log` to blend the logs together.  Alternatively we can probably change the [`$ibm_cloud_log` initialization](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/17faf6d2fd99e35059b4dc7637ade4861927c9fb/lib/manageiq/providers/ibm_cloud/engine.rb#L22-L24) to also broadcast to `$log` much in the same way we do for the container logger, though we'll have to make sure we aren't "double-logging".  On that note, if this is something that we think would benefit all providers, we can also just do that in the core logger initialization directly.  @agrare Thoughts on this suggestion since you are familiar with the other providers?

Reply:
I don't mind removing the switch as long as it is verified that the provider logging will show up in pods. As I said the only reason I did that was because I couldn't find my logging messages in pods. The other part, I'm keeping in. The logging is much simpler and standardized.

Reply:
> verified that the provider logging will show up in pods.

```ruby
> CONTAINER=true bin/rails c
...
[1] pry(main)> $ibm_cloud_log.info("hi")
{"@timestamp":"2021-04-30T14:28:00.880548 ","pid":14728,"tid":"9164","level":"info","message":"hi"}
```

> The other part, I'm keeping in. The logging is much simpler and standardized.

Not sure I follow.  From what I can tell it's not following the standardization provided by the LogProxy, because it builds an [alternative string](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/198/files#diff-f7024315333d99e0f6101d4f8bdcef8ec7e6c162cf350548f6bf27b26fa2b452R83).